enhancements/proposals/2549-scylladbcluster-backup-repair/README.md

```diff
@@ -0,0 +1,589 @@
+    The ScyllaDBManagerTask controller propagates the value of the annotation on task update/creation with ScyllaDB Manager.
+    The field, as well as support for `TZ=` and `CRON_TZ=` prefixes in `Cron`, can be introduced once [scylladb/scylla-manager#3818][1] is resolved. For now, we make the validation as strict as possible.
+  - The default value for `NumRetries` is removed from the new API. The defaults for not provided parameters should be handled by ScyllaDB Manager server.
+- Backup:
+  - The default value for `Retention` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+- Repair:
+  - The type of `SmallTableThreshold` field is changed to `*resource.Quantity` to align with in-tree Kubernetes objects. Validation will enforce non-decimal values.
+  - The type of `Intensity` field is changed to `int64` to align with ScyllaDB Manager server.
+    ScyllaDB Manager API server accepts a float64 value, but overrides any values in (0,1) range. It responds with a 400 status on decimal values > 1. Unfortunately, ScyllaCluster validation allows for decimal values > 1.
+    As we can not make the validation stricter on translation, to maintain backwards compatibility, the translating controller will set an internal annotation `internal.scylla-operator.scylladb.com/scylladb-manager-task-repair-intensity-override` on applied ScyllaDBManagerTask objects.
+    ScyllaDBManagerTask controller will use it to override the value set in the object's specification.
+  - The default value for `FailFast` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+  - The default value for `SmallTableThreshold` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+  - The default value for `Retention` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+  - The default value for `Parallel` is removed from the new API. The ScyllaDB Manager server assumes a default value if the parameter is no set explicitly.
+  - ScyllaDBManagerTask controller will no longer override `NumRetries` when `FailFast` is set to true. If required, such behaviour should be enforced by ScyllaDB Manager server.
+    The translating controller will maintain this behaviour when applying ScyllaDBManagerTasks.
+
+The translating controller will keep respecting the default values defined in ScyllaCluster API and propagate them to applied ScyllaDBManagerTasks.
+
+#### Finalizers
+
+The new ScyllaDBManagerTask controller will set a `scylla-operator.scylladb.com/scylladbmanagertask-protection` finalizer on ScyllaDBManagerTask objects so that the tasks are deleted from ScyllaDB Manager state on object deletion.
```

Comment:
"protection" will mean nothing to the user. If deletion gets stuck on "protection", they'll be absolutely clueless.

I suggest using a name that will allow the user to infer the meaning. For example:

```suggestion
The new ScyllaDBManagerTask controller will set a `scylla-operator.scylladb.com/scylladbmanagertask-ensure-deregistration` finalizer on ScyllaDBManagerTask objects so that the tasks are deleted from ScyllaDB Manager state on object deletion.
```

Reply:
I was trying to stay consistent with existing RemoteKubernetesCluster and ScyllaDBCluster finalizers, but that's a fair point.
Would `scylla-operator.scylladb.com/scylladbmanagertask-deregistration` suffice though?

Reply:
absolutely :+1: thanks!

Reply:
changed to `scylla-operator.scylladb.com/scylladbmanagertask-deletion`, as that's the operation performed with manager